rpc/js/src/dial.ts

```diff
@@ -59,43 +90,51 @@ async function makeAuthenticatedTransportFactory(address: string, defaultFactory
 	const getExtraMetadata = async (): Promise<grpc.Metadata> => {
 		// TODO(GOUT-10): handle expiration
 		if (accessToken == "") {
-			const request = new AuthenticateRequest();
-			request.setEntity(opts?.authEntity ? opts.authEntity : address.replace(/^(.*:\/\/)/, ''));
-			const creds = new PBCredentials();
-			creds.setType(opts?.credentials?.type!);
-			creds.setPayload(opts?.credentials?.payload!);
-			request.setCredentials(creds);
+			let thisAccessToken = "";
 
 			let pResolve: (value: grpc.Metadata) => void;
 			let pReject: (reason?: unknown) => void;
-			let done = new Promise<grpc.Metadata>((resolve, reject) => {
-				pResolve = resolve;
-				pReject = reject;
-			});
-			let thisAccessToken = "";
-			grpc.invoke(AuthService.Authenticate, {
-				request: request,
-				host: opts?.externalAuthAddress ? opts.externalAuthAddress : address,
-				transport: defaultFactory,
-				onMessage: (message: AuthenticateResponse) => {
-					thisAccessToken = message.getAccessToken();
-				},
-				onEnd: (code: grpc.Code, msg: string | undefined, _trailers: grpc.Metadata) => {
-					if (code == grpc.Code.OK) {
-						pResolve(md);
-					} else {
-						pReject(msg);
+
+			if (opts?.webrtcOptions?.signalingAccessToken != "") {
```

Comment:
this is almost right but in makeAuthenticatedTransportFactory, we don't want to know about webrtcOptions, just access tokens. That's why you see field reassignment in dialWebrtc. As far as both direct dialing and webrtc signaling are concerned, the authentication process is the same. If you are dialing direct and have a token, we use that token instead of the creds, but still call `AuthenticateTo` in the event external addr is provided.

For signaling, the same logic applies, we will directly pass the access token but if the external address is set, we will pass the token to that external address and call `AuthenticateTo`.

Overall, that means this function should never check a `opts.webrtcOptions`

Reply:
Updated to pass accessToken to the AuthenticateTo() request if the external entity and other option are set.